packages/@idemsInternational/gdrive-tools/src/commands/authorize.ts

```diff
@@ -148,27 +152,52 @@ function getAccessToken(
   tokenPath: string,
   callback: (client: OAuth2Client) => void
 ) {
+  // Generate the url that will be used for the consent dialog.
   const authUrl = oAuth2Client.generateAuthUrl({
     access_type: "offline",
     scope: scopes,
   });
-  console.log("Authorize this app by visiting:\n");
-  console.log("\x1b[36m" + authUrl + "\n" + "\x1b[0m");
-  const rl = readline.createInterface({
-    input: process.stdin,
-    output: process.stdout,
-  });
-  rl.question("Enter the code from that page here: ", (code) => {
-    rl.close();
-    oAuth2Client.getToken(code, (err, token) => {
-      if (err) return console.error("Error retrieving access token", err);
-      oAuth2Client.setCredentials(token);
-      // Store the token to disk for later program executions
-      fs.writeFileSync(tokenPath, JSON.stringify(token));
-      console.log("Token stored to", tokenPath);
-      callback(oAuth2Client);
+  console.log(
+    chalk.yellow(
+      "Authorize this app in your browser. If not automatically redirected, please manually visit:\n"
+    )
+  );
+  console.log(chalk.blue(authUrl + "\n"));
+
+  // Open an http server to accept the oauth callback.
+  const server = http
+    .createServer(async (req, res) => {
+      try {
+        // The request endpoint of "/oauth2callback" is configured
+        // within Google Cloud's API Console Credentials page
+        if (req.url.indexOf("/oauth2callback") > -1) {
+          const qs = new url.URL(req.url, "http://localhost:8080").searchParams;
+          res.end(
```

Comment:
does the callback have to be served on port 8080 or can it be any port? If it can be any port I'd probably suggest picking one less likely to be in use by other programs. I can see the docs using port 3000 (would clash with react apps but should be fine for angular), or can just pick a more random one